addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -241,12 +251,18 @@ class dragAndDropHelper {
                 this._removeGridAndDragHelper(rowEl);
                 const rowCount = parseInt(rowEl.dataset.rowCount);
                 dropzoneEl.style.gridRowEnd = Math.max(rowCount + 1, 1);
+                this._unwrapDraggedItemIfNeeded();
             }
 
-            this.draggedItemEl.remove();
```

Comment:
This has been moved before `this._unwrapDraggedItemIfNeeded();` as there could be a change of `this.draggedItemEl`